CadbaseLibrary/CadbaseLibrary.FCMacro

```diff
@@ -0,0 +1,837 @@
+                        os.rmdir(new_info_file)
+                    else:
+                        os.remove(new_info_file)
+
+                f = open(new_info_file, "x")
+                data = json.dumps(modification, default=lambda o: o.__dict__, indent=4)
+                f.write(data)
+                f.close()
+                # FreeCAD.Console.PrintWarning(new_info_file + '\n')
+    FreeCAD.Console.PrintWarning('Finished\n')
+
+def updateComponentModificaion(LIBRARYPATH, modification_path, modification_uuid):
+
+    # FreeCAD.Console.PrintWarning('Getting files\n')
+
+    data = parsingGpl(LIBRARYPATH)
+    if isinstance(data, SimpleNamespace):
+
+        if len(data.componentModificationFilesets) == 0:
+
+            FreeCAD.Console.PrintWarning('Fileset not found for FreeCAD\n')
+
+        else:
+
+            CDBSApi(QueriesAPI.filesetFiles(data.componentModificationFilesets[0].uuid), LIBRARYPATH)
+
+            data = parsingGpl(LIBRARYPATH)
+            if isinstance(data, SimpleNamespace):
+
+                if len(data.componentModificationFilesetFiles) == 0:
+                    FreeCAD.Console.PrintWarning('No files in fileset\n')
+
+                else:
+
+                    urls = []
+                    fns = []
+
+                    for file_of_fileset in data.componentModificationFilesetFiles:
+                        urls.append(file_of_fileset.downloadUrl)
+                        fns.append(modification_path + "/" + file_of_fileset.filename)
+
+                    inputs = zip(urls, fns)
+                    download_parallel(inputs)
+                    FreeCAD.Console.PrintWarning('Download is completed (')
+                    FreeCAD.Console.PrintWarning(len(urls))
+                    FreeCAD.Console.PrintWarning(' file(s)\n')
+
+    FreeCAD.Console.PrintWarning('Finished\n')
+
+def download_parallel(args):
+    # FreeCAD.Console.PrintWarning('Start download data\n')
+
+    t0 = time.time()
+
+    results = ThreadPool(cpu_count() - 1).imap_unordered(get_file, args)
+
+    for result in results:
+        FreeCAD.Console.PrintWarning('\npath: ')
+        FreeCAD.Console.PrintWarning(result[0])
+        FreeCAD.Console.PrintWarning(' time (s): ')
+        FreeCAD.Console.PrintWarning(result[1])
+
+    FreeCAD.Console.PrintWarning('\nTotal time:')
+    FreeCAD.Console.PrintWarning(time.time() - t0)
+    FreeCAD.Console.PrintWarning('\n')
+
+def get_file(args):
+    t0 = time.time()
+    url, filepath = args[0], args[1]
+    manager = QtNetwork.QNetworkAccessManager()
+
+    try:
+        request = QtNetwork.QNetworkRequest(QtCore.QUrl(url))
+        # request.setRawHeader(b"Content-Type", b"application/json");
+        request.setRawHeader(b'User-Agent', user_agent)
+
+        reply = manager.get(request)
+        loop = QtCore.QEventLoop()
+        reply.finished.connect(loop.quit)
+        loop.exec_()
+
+        if reply.error() == QtNetwork.QNetworkReply.NoError:
+
```

Comment:
Please reduce the number of empty lines.

Reply:
Fixed, but maybe I missed something because of bad style.